File: app/src/main/java/xyz/yakdmt/followmeradio/data/DataManager.java
package xyz.yakdmt.followmeradio.data;

import java.util.concurrent.Executors;
import java.util.concurrent.ScheduledExecutorService;
import java.util.concurrent.TimeUnit;

import javax.inject.Inject;

import xyz.yakdmt.followmeradio.network.NetworkWorker;

/**
 * Created by yakdmt on 11/06/16.
 */
public class DataManager {

    private final NetworkWorker mNetworkWorker;
    private final ScheduledExecutorService;

    @Inject
    DataManager(NetworkWorker networkWorker) {
        mNetworkWorker = networkWorker;
        ScheduledExecutorService scheduler =
                Executors.newSingleThreadScheduledExecutor();

        scheduler.scheduleAtFixedRate
                ((Runnable) () -> {
                    // call service
                }, 0, 10, TimeUnit.SECONDS);
    }


}

File: app/src/main/java/xyz/yakdmt/followmeradio/data/DataManager.kt
package xyz.yakdmt.followmeradio.data

import org.greenrobot.eventbus.EventBus
import javax.inject.Inject

import xyz.yakdmt.followmeradio.network.NetworkWorker
import xyz.yakdmt.followmeradio.utils.Event

/**
 * Created by yakdmt on 20/04/16.
 */
class DataManager
@Inject
constructor(private val mNetworkWorker: NetworkWorker) {
    init {
        EventBus.getDefault().register(this);
    }
}

@SuppressWarnings("unused")
fun onEvent(event: Event.OnAudioStateChanged) {

}

